core/domain/state_domain.py

```diff
@@ -1910,29 +1910,23 @@ def update_interaction_hints(self, hints_list):
         self._update_content_ids_in_assets(
             old_content_id_list, new_content_id_list)
 
-    def update_interaction_solution(self, solution_dict):
+    def update_interaction_solution(self, solution):
         """Update the solution of interaction.
 
         Args:
-            solution_dict: dict or None. The dict representation of
-                Solution object.
+            sol: Solution object.
```

Comment:
Use the actual variable name and mention the type as 'Solution'. Something like `solution: Solution. The solution object.`